### PR TITLE
feat: [83] Create PlannedTransaction model and migration

### LINKED ISSUE #83: Create PlannedTransaction model and migration
### Overview
Create a separate `PlannedTransaction` model for recurring/planned transaction templates. These are not actual transactions — they're templates that generate virtual calendar entries across future dates.

### Why a Separate Model?
A planned transaction represents a *template* that generates virtual calendar entries across many future dates. It has its own lifecycle (activate/deactivate, change frequency, set end date) separate from actual bank transactions. The two models connect through `planned_transaction_id` on `Transaction` when reconciliation happens.

### Migration: `create_planned_transactions_table`

| Column | Type | Notes |
|--------|------|-------|
| `id` | bigIncrements | |
| `user_id` | foreignId | constrained, cascadeOnDelete |
| `account_id` | foreignId | constrained, cascadeOnDelete |
| `category_id` | foreignId | nullable, nullOnDelete |
| `amount` | bigInteger | Cents, via MoneyCast |
| `direction` | string | TransactionDirection enum |
| `description` | string | |
| `start_date` | date | First occurrence |
| `frequency` | string | RecurrenceFrequency enum |
| `until_date` | date | nullable; null = repeats forever |
| `is_active` | boolean | default true |
| `last_generated_date` | date | nullable; tracks materialized occurrences |
| timestamps | | |

Indexes: `[user_id, start_date]`, `[user_id, is_active]`

### Model: `app/Models/PlannedTransaction.php`

Relationships:
- `belongsTo User`
- `belongsTo Account`
- `belongsTo Category` (nullable)
- `hasMany Transaction` (reconciled actuals via `planned_transaction_id`)

Key method:
- `occurrencesBetween(CarbonImmutable $start, CarbonImmutable $end): Collection` — generates date occurrences using `RecurrenceFrequency::nextOccurrence()` iteratively

### Factory: `database/factories/PlannedTransactionFactory.php`
States: `weekly()`, `monthly()`, `noRepeat()`, `withEndDate()`

### Tests
- `tests/Feature/Models/PlannedTransactionTest.php`
- `occurrencesBetween()` returns correct dates for each frequency
- Respects `until_date` boundary
- Respects `is_active` flag
- Edge cases: start_date after range, monthly on 31st, empty range

### Acceptance Criteria
- [ ] Migration creates table with all columns and indexes
- [ ] Model has typed relationships with return type hints
- [ ] `occurrencesBetween()` works correctly for all frequencies
- [ ] Factory with useful states
- [ ] Feature tests pass
- [ ] `op ci` passes

### Dependencies
- Issue #1 (TransactionDirection enum — already exists)
- Issue #2 (RecurrenceFrequency enum — needed for `frequency` cast)

---
Part of the **Add/Edit Transaction Modal** feature.

### PR DESCRIPTION
**Pull request by robwilde**
## Summary
- Create `PlannedTransaction` model — a recurring/planned transaction template that generates virtual calendar entries across future dates, separate from actual `Transaction` records
- Add migration with full schema (columns, indexes, FK constraints) including a follow-up migration to add the foreign key constraint for `planned_transaction_id` on the existing `transactions` table
- Implement `occurrencesBetween(CarbonImmutable $start, CarbonImmutable $end)` method that iterates using `RecurrenceFrequency::nextOccurrence()` to generate date occurrences within a range
- Add factory with `weekly()`, `monthly()`, `noRepeat()`, `withEndDate()`, and `inactive()` states
- Wire up relationships: `Transaction.plannedTransaction()` BelongsTo, `User.plannedTransactions()` HasMany

## Test plan
- [x] 36 feature tests covering factory defaults, relationships, cascade behavior, casts, factory states, and `occurrencesBetween()` logic
- [x] `occurrencesBetween()` edge cases: `DontRepeat`, `until_date` boundary, inactive flag, start before/after range, monthly on 31st (short months), workday skipping weekends
- [x] `op migrate.fresh` runs clean
- [x] `op ci` passes (lint, PHPStan, 644 tests)

Closes #83

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: database/factories/PlannedTransactionFactory.php [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    /**
+     * @return array<string, mixed>
+     */
+    public function definition(): array
+    {
+        $user = User::factory();
+
+        return [
+            'user_id' => $user,
+            'account_id' => Account::factory()->for($user),
```
[the remark this answers is not in the document]
Comment (robwilde) [reply database/factories/PlannedTransactionFactory.php:28]: Not a bug — this is a false positive. Laravel resolves the `User::factory()` instance once and reuses it for both `user_id` and `Account::factory()->for($user)`. This is the exact same pattern used in `TransactionFactory` (line 26-28) which is validated by existing tests. No change needed.

File: app/Models/PlannedTransaction.php [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        if (! $this->is_active) {
+            return collect();
+        }
+
+        $dates = collect();
+        $current = $this->start_date;
+        $limit = $this->until_date;
+        $maxIterations = 1000;
+
+        if ($this->frequency === RecurrenceFrequency::DontRepeat) {
```
[the remark this answers is not in the document]
Comment (robwilde) [reply app/Models/PlannedTransaction.php:92]: Valid observation. The 1000-iteration guard is intentional — this method is designed for calendar views (typically 1-12 months), and 1000 iterations covers daily recurrence for ~2.7 years which is more than adequate. The main risk is an old start_date needing many iterations to reach the range; a skip-ahead optimisation can be added in a future issue if longer horizons are needed.

File: database/seeders/PlannedTransactionSeeder.php [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+declare(strict_types=1);
+
+namespace Database\Seeders;
+
+use Illuminate\Database\Seeder;
+
+final class PlannedTransactionSeeder extends Seeder
+{
+    public function run(): void {}
+}
```
[the remark this answers is not in the document]
Comment (robwilde) [reply database/seeders/PlannedTransactionSeeder.php:12]: Intentional — matches project convention. TransactionSeeder is also an empty stub. These are scaffolded placeholders for future seeding logic.

File: tests/Feature/Models/PlannedTransactionTest.php [hunk 4]
```diff
@@ -0,0 +1,402 @@
...
+    expect($planned->category_id)->toBeNull();
+});
+
+test('planned transaction has many transactions', function () {
+    $planned = PlannedTransaction::factory()->create();
+    Transaction::factory()->count(3)
+        ->for($planned->user)
+        ->for($planned->account)
+        ->create(['planned_transaction_id' => $planned->id]);
+
```
[the remark this answers is not in the document]
Comment (robwilde) [reply tests/Feature/Models/PlannedTransactionTest.php:71]: Consistent with existing project convention — TransactionTest uses the same raw FK pattern (e.g. create(['transfer_pair_id' => $pair->id]) on line 208). No change needed.

File: tests/Feature/Models/PlannedTransactionTest.php [hunk 5]
```diff
@@ -0,0 +1,402 @@
...
+        ->each(fn (Pest\Expectation $transaction) => $transaction->toBeInstanceOf(Transaction::class));
+});
+
+test('transaction belongs to planned transaction', function () {
+    $planned = PlannedTransaction::factory()->create();
+    $transaction = Transaction::factory()
+        ->for($planned->user)
+        ->for($planned->account)
+        ->create(['planned_transaction_id' => $planned->id]);
+
```
[the remark this answers is not in the document]
Comment (robwilde) [reply tests/Feature/Models/PlannedTransactionTest.php:82]: Same as above — matches existing test patterns in the codebase.
